pkg/kvdpa/device.go

```diff
@@ -185,44 +185,30 @@ func GetVdpaDevice(name string) (VdpaDevice, error) {
 		return nil, err
 	}
 
-	vdpaDevs, err := parseDevLinkVdpaDevList(msgs)
+	// No filters, expecting to parse attributes for the device with the given name
+	vdpaDevs, err := parseDevLinkVdpaDevList("", "", msgs)
 	if err != nil {
 		return nil, err
 	}
 	return vdpaDevs[0], nil
 }
 
-/*
-GetVdpaDevicesByMgmtDev returns the VdpaDevice objects whose MgmtDev
-has the given bus and device names.
-*/
-func GetVdpaDevicesByMgmtDev(busName, devName string) ([]VdpaDevice, error) {
-	result := []VdpaDevice{}
-	devices, err := ListVdpaDevices()
+// ListVdpaDevices returns a list of all requested vdpa devices
+// fullMgmtDeviceName (if not empty) will act as a filter
+// Example1: "pci/0000:01:02" will return all the devices having bus=pci and mgmtdev=0000:01:02
+// Example2: "0000:01:02" will return all the devices having mgmtdev=0000:01:02 (no bus selected)
+func ListVdpaDevices(fullMgmtDeviceName string) ([]VdpaDevice, error) {
```

Comment:
This introduces unnecessary API breakage. You achieve the same (with same level of code reuse) with:
```go
func listVdpaDevices(busName, devName string) ([]VdpaDevice, error) {} // Private!

func ListVdpaDevices()  ([]VdpaDevice, error) { return listVdpaDevices("", "") }
func GetVdpaDevicesByMgmtDev (busName, devName string) ([]VdpaDevice, error) { return listVdpaDevices(busName, devName) }
```

Reply:
ovn-k is the only one using a govdpa public function, i.e. `GetVdpaDevicesByPciAddress`. I don't think `GetVdpaDevicesByMgmtDev` is used anywhere. I don't remember why I have introduced `GetVdpaDevicesByPciAddress` in the first place though, it seems a duplication of `GetVdpaDevicesByMgmtDev` . Anyway ovn-k is currently using govdpa version `v0.1.4` not the latest, so shouldn't this prevent any backward compatiblity issues?

Reply:
Agree, `GetVdpaDevicesByPciAddress` should go, let's create an issue for that, but it's still unnecessary to change `ListVdpaDevices` or `GetVdpaDevicesByPciAddress`. The API breakage would force us to bump a mayor version with no strong reason.

Also, ovn-k is not the only user of govdpa

Reply:
- `GetVdpaDevicesByPciAddress(pciAddress string)` is used by ovn-k, so we shouldn't remove it
- `GetVdpaDevicesByMgmtDev(busName, devName string)` needs to stay for compatibility
- `ListVdpaDevices()` needs to stay for compatibility
- `ListVdpaDevices(fullMgmtDeviceName string)` will become private and CLI will use one of the above

When bumping a new release we'll do the cleaning, do you agree?

Reply:
Yes, API should remain untouched for this PR